lib/ansible/modules/cloud/vmware/vca_vapp.py

```diff
@@ -229,6 +234,55 @@ def set_state(module):
         if not vapp.undeploy(action):
             module.fail('unable to undeploy vapp')
 
+def connect_to_network(module, vdc_name, vapp_name, network_name, network_mode):
+
+    nets = filter(lambda n: n.name == network_name, module.vca.get_networks(vdc_name))
+    assert len(nets) == 1
+    the_vdc = module.vca.get_vdc(vdc_name)
+    the_vapp = module.vca.get_vapp(the_vdc, vapp_name)
+    assert the_vapp
+    assert the_vapp.name == vapp_name
+
+    # Connect vApp
+    task = the_vapp.connect_to_network(nets[0].name, nets[0].href)
+    result = module.vca.block_until_completed(task)
+    assert result
+
+    # Connect VM
+    if(network_mode == 'pool'):
```

Comment:
For this module there are 3 network types supported:
* static
* pool
* dhcp

Looking at the pyvcloud.vapp library it appears that the supported modes are:
* pool
* dhcp
* manual
* none

I tested changing from static (which is what matches the UI) to pool, and it appears to have worked.  This block doesn't do anything if network_mode is set to static, which is why mine was failing i think.

You may have already covered this in your upcoming patch but i figured it would be good to point it out.